File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string textBlock = get_string("Text: ");
    int characters = strlen(textBlock);
    int sentences = 0;
    int spaces = 0;
    int letters = characters;
    for (int i = 0; i < characters; i++)
    {
        if (textBlock[i] == 32) // space
        {
            spaces++;
            letters--;
        }
        else if (textBlock[i] == 46 || textBlock[i] == 33 || textBlock[i] == 63) // period, exclamation mark, question mark
        {
            sentences++;
            letters--;
        }
        else if (textBlock[i] <65 || textBlock[i] > 122 || (textBlock[i] > 90 && textBlock[i] < 97)) // other punctuation or numbers
        {
            letters--;
        }
    }

    int words = spaces + 1;

    // calculate L (letters/(words/100))
    // where L is the average number of letters per 100 words in the text,
    float L = (float) letters / ((float) words/100);

    // calculate S (sentences/(words/100))
    // where S is the average number of sentences per 100 words in the text.
    float S = (float) sentences / ((float)words/100);

    // Finally, calculate index
    float index = round(0.0588 * L - 0.296 * S -15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
    return 0;
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0
for char in text:
    if (char == " "):
        words = words + 1
    if (char == "." or char == "!" or char == "?"):
        sentences = sentences + 1
    if ((ord(char) >= 65 and ord(char) <= 90) or (ord(char) >= 97 and ord(char) <= 122)):
        letters = letters + 1
# print(f"letters= {letters}")
# print(f"words= {words}")
# print(f"sentences= {sentences}")
# print(f"letters per 100 words = {letters/words*100}")
# print(f"sentences per 100 words = {sentences/words*100}")

grade = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8

if (grade < 1):
    print("Before Grade 1")
elif (grade >= 16):
    print("Grade 16+")
else:
    print(f"Grade {round(grade)}")
